### PR TITLE
Domains: Add Notice if User Already Owns a Custom Domain

### LINKED ISSUE #27932: Add an alert message during the domains purchase flow if the user already has a domain associated with their site
As a new Happiness Engineer for WP.com, there's a situation I keep running into repeatedly. This week alone, I've helped three different users who have made this mistake. They register a new domain associated with their existing site (which already has a domain), when what they actually meant to do was create a new site and register a domain for the new site. They think adding a new domain is how they create a new site.

It can sometimes be time consuming to help the user get everything sorted out, show them how to create a new site, and get the domain associated with the new site. We also typically have to issue a refund as well.

I'd like to make a suggestion to add a notification or alert message somewhere during the domains purchase flow, if the user _already_ has a domain associated with their site.

The additional domain is purchased via My Site > Domains > Add.

I think having it pop up at the beginning of the process, like on this screen, would be ideal: http://cld.wthms.co/42m8vc

The message could say something along the lines of:
>You're purchasing an additional domain for CURRENT SITE. Would you like to create a new site instead? If so, click here.

I believe this would clear up a lot of confusion :)

### LINKED ISSUE #39829: Domains: Lacking intro message when adding a new domain
#### Steps to reproduce
1. Go to your **My Site** on a site that already has a custom primary domain assigned to it
2. Click on **Domains > Add**
3. You will be offered the possibility to register or map a domain, with **no additional text of explanation**.

#### What I expected
I would expect a **short text introduction** clarifying that I am about to add a new domain name for an existing site, as opposed to registering a new site.

#### What happened instead
There is no message, explanation or links to the documentation letting the user what they are about to do and offering a link to the "**Register a new site**" flow should they be looking for that other option (as it is often the case. 

I find this a lot in chat and tickets. **A user intends to register a new (different) site buy will instead add a secondary domain to their current page**, which can be rather confusing. A simple text with a couple of links will reduce frustration, confusion and support time.

#### Browser / OS version
MacOS 10.13.6 (17G2307)
Google Chrome 79.0.3945.88

#### Screenshot / Video
I made this mockup.

domain-search-e280b9-julia-e28094-wordpress-com

The text could go along these lines:

> Here is the corrected version of the text we could use (or something to that effect)
> 
> You are about to register a new domain for CurrentPrimaryDomain. This domain will be added to your site SiteAddress. This is not the same as creating a new site. If you want to create a separate new site please click here.
> 
> You can click here to learn more about domains.

#### Context / Source
This is a rather common recurring issue:
h16476864-hc 16478017-hc 16479872-hc

Also, discussed here: p9ag8j-Vh-p2

### PR DESCRIPTION
**Pull request by Aurorum**
#### Changes proposed in this Pull Request

* Adds a notice if the user already has a custom domain and is trying to register a new one. There's some more detailed context in the original issue, but basically a significant number of people actually intend to create a new site, but end up registering a domain for their current site

#### Testing instructions

Visit `/domains/add/` on a site with a custom domain, and verify that the notice appears. It should not appear if you don't have a custom domain.

Screenshot 2019-07-31 at 08 31 57

@michelleweber, is this wording okay?

> You're purchasing an additional domain for (site title). Would you like to create a new site instead?
> Create a new site

Fixes #27932 and fixes #39829

cc @sgomes, @jsnajdr, @aidvu

### CODE REVIEW

File: client/my-sites/domains/domain-search/index.jsx
Comment (a8ci18n) [inline client/my-sites/domains/domain-search/index.jsx]: Hi! I've found a possible matching string that has already been translated **33** times:
`translate( 'Create New Site' )` *ES Score: 7*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/pull/34990)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/my-sites/domains/domain-search/index.jsx
Comment (aidvu) [inline client/my-sites/domains/domain-search/index.jsx]: This should send users to `/start` (NUX) rather than a `/jetpack/` flow.

**Review by aidvu — COMMENTED** [review]
Added @Automattic/cobalt as reviewers. 

Also, thanks for working on this @Aurorum. 🙇

File: client/my-sites/domains/domain-search/index.jsx
Comment (Aurorum) [reply client/my-sites/domains/domain-search/index.jsx]: Updated! My only concern with that URL though is that, unlike the previous one, users lose access to the Back button. It's also the link people are redirected to when clicking "Add New Site". 

Screenshot 2019-07-30 at 13 29 24

Comment (aidvu) [reply client/my-sites/domains/domain-search/index.jsx]: > My only concern with that URL though is that, unlike the previous one, users lose access to the Back button. It's also the link people are redirected to when clicking "Add New Site".

I see. Not sure if we have an easy way to do that with `/start`.

Comment (Aurorum) [reply client/my-sites/domains/domain-search/index.jsx]: It looks like the "Back to My Sites" link from `/start` was originally added under the assumption that there wouldn't be a case like this. From @sixhours in #29952:

> Taking a step back and looking at this again, **we're assuming the user has clicked on My Sites, switched sites, and opted to add a new site**. Therefore, sending them back to /sites makes sense to me, and we can use the (already familiar) language "My Sites", which is what they'll see at that location.

Screenshot 2019-07-30 at 14 37 14

The solutions probably are:

1. Just don't offer a Back link for this flow
2. Change "Back to My Sites" into an ordinary "Back" link
3. Switch this link back to `/jetpack/new?ref=calypso-selector` again

I was inclined to go with two, but even that's a bit tricky to implement as `/start` redirects a lot. Maybe `/jetpack/new?ref=calypso-selector` is the better link to go with after all?

Comment (aidvu) [reply client/my-sites/domains/domain-search/index.jsx]: At this point I'm inclined to see what the @Automattic/zelda thinks.

Comment (ramonjd) [reply client/my-sites/domains/domain-search/index.jsx]: > Maybe /jetpack/new?ref=calypso-selector is the better link to go with after all?

I'm inclined to agree with this, only because it matches the logged-in behaviour of **+ ADD NEW SITE** (in the side bar) and it keeps things simple.

The link goes here.

Screen Shot 2019-07-31 at 10 28 06 am

It might not be ideal as it introduces the notion of 'Jetpack' to the user, which might be confusing, but @aidvu is right: the signup flow messes with the route history. It shunts the user along to the most appropriate step, which is, in most cases `/start/site-type`. 

In this situation, using the  `getPreviousRoute` selector, which might have been a possible solution to the back button url dilemma, will return `/start` and not `/domains/add/{site}`

**TL;DR**  The path of less resistance is to use `/jetpack/new?ref=calypso-selector`

One more thing, in my opinion, it should be `'Create a new site'`, with the indefinite article.

Comment (Aurorum) [reply client/my-sites/domains/domain-search/index.jsx]: I also ran into that issue when trying to use `getPreviousRoute`, updated the URL to `/jetpack/new?ref=calypso-selector` now. Also added the indefinite article, thanks @ramonjd! :)

**Review by aidvu — DISMISSED** [review]
@ramonjd this is LGTM now. You ok with it?

File: client/my-sites/domains/domain-search/index.jsx
Comment (aidvu) [inline client/my-sites/domains/domain-search/index.jsx]: I now see this. No need to split them. Please make it a single string. :)

**Comment by wensco** [comment]
Can we hold off on this change until we can discuss further please? With all due respect, I believe we are trying to fix the original problem (users can't figure out how to create a new site) the wrong way.  :)

Instead of adding messaging and possibly dissuading people from buying domains who actually want to buy a domain, we should do a better job of surfacing the option to create a new site.

I've personally experienced the issue since the option is completely invisible unless you choose the site switcher. Now, if a user only has one site, I can see why it would not occur to them to ever use the switcher. Even if they have multiple sites, using a switcher to create a new site is really not intuitive.

As a side note, I searched our user support docs for "create a new site" and "add a site" and could not find anything relevant so that's an opportunity as well.

**Comment by Aurorum** [comment]
> Can we hold off on this change until we can discuss further please?

:+1:

> Now, if a user only has one site, I can see why it would not occur to them to ever use the switcher.

I don't believe that users with one site have access to the switcher? "Add New Site" is constantly present for them. 

add-new-site

> As a side note, I searched our user support docs for "create a new site" and "add a site" and could not find anything relevant so that's an opportunity as well.

Yep, there's already a support document for them, it'd probably just be helpful to include it in those searches: https://en.support.wordpress.com/create-a-blog/

File: client/my-sites/domains/domain-search/index.jsx
Comment (Aurorum) [reply client/my-sites/domains/domain-search/index.jsx]: Would this not run the risk of translators missing the space at the end of each sentences? I recall that happening before and since then, I've always split them. :)

Comment (aidvu) [reply client/my-sites/domains/domain-search/index.jsx]: Not really. This is a single message and should be translated as such. We have plenty of these, even in this file: https://github.com/Automattic/wp-calypso/blob/b349ba2c29137398527ea5e1864bd64cf408681b/client/my-sites/domains/domain-search/index.jsx#L151

Comment (Aurorum) [reply client/my-sites/domains/domain-search/index.jsx]: Ah, I understand now. Updated, thanks! :)

**Comment by ramonjd** [comment]
> this is LGTM now. You ok with it?

Thanks @aidvu 

I'm okay with it in terms of code changes. 

As for the UX/business points @wensco raises, I'll defer to those with more context than I have right now.

**Comment by Aurorum** [comment]
#36637 also suggested the option of increasing visibility of the option to add a new site. 

@wensco What do you think of adding an "Add New Site" button to the list of buttons that appear at `/domains/manage/site`, and rephrasing the primary button's text if the user has a custom domain?

Screenshot 2019-10-10 at 17 11 44

**Comment by wensco** [comment]
> @wensco What do you think of adding an "Add New Site" button to the list of buttons that appear at `/domains/manage/site`, and rephrasing the primary button's text if the user has a custom domain?
Thanks for the ping! I commented on this related issue [here](https://github.com/Automattic/wp-calypso/issues/27932#issuecomment-548590229).

tl;dr I think this is more subtle but functionally the same as the banner. We're also conflating sites and domains this way and they are not the same thing. Why not expose the "create site" function better in the sidebar? The root issue here I believe is that users can't figure out how to create a new site. :)

**Comment by sixhours** [comment]
Hallo! I'm going through the stuff that fell through the cracks again. I think this was eventually handled in a different way by this "Add a domain" button:
Screen Shot 2021-05-28 at 2 34 21 PM

Going to close this out, but as always, thanks for your work Aurorum! <3

**Pull request closed without merging**
